### PR TITLE
Merge MOL 19 and 37 into dev

### PR DESCRIPTION
**Pull request by kdm5498**
Provides a proxy API to integrate the Autodock and OpenBabel cloud deployments.

Provides 2 additional API methods:
dockLigand - POST - Accepts a ligand and macromolecule file, converts the files to pdbqt format, and passes the files to the Autodock API, returning a Job ID.
dockLigand - GET - Accepts an Autodock Job ID and will either return the status of the job, or the completed job.

### CODE REVIEW

**Comment by kdm5498** [comment]
There are some conflicts with the existing dev branch, but we should be able to just accept my changes, I already merged dev into this branch and resolved the conflicts.

File: src/main/java/org/moltimate/moltimatebackend/dto/request/DockingRequest.java [hunk 1]
```diff
@@ -3,18 +3,46 @@
...
+
+	public MultiValueMap<String, Object> toMap() throws IOException {
+		MultiValueMap<String, Object> map = new LinkedMultiValueMap<>();
+		map.add("macromolecule", new ByteArrayResource( macromolecule.getBytes() ){
+			@Override
+			public String getFilename() {
+				return macromolecule.getOriginalFilename();
+			}
+		});
+		map.add("ligend", new ByteArrayResource( ligand.getBytes() ){//TODO: Fix spelling when API is fixed
```
Comment (tydus101) [inline src/main/java/org/moltimate/moltimatebackend/dto/request/DockingRequest.java]: I know that I like to spell ligand wrong but we should probably commit code with the correct spelling :P

File: src/main/java/org/moltimate/moltimatebackend/service/DockingService.java [hunk 2]
```diff
@@ -2,13 +2,208 @@
...
+		String babelHash = template.postForEntity( openBabelURL, entityBabelPost, String.class ).getBody();
+
+		// Check the status of the file conversion every 20 seconds.
+
+		try {
+			ResponseEntity<byte[]> babelConversion =
+					template.getForEntity( openBabelURL + "?storage_hash=" + babelHash, byte[].class );
+			while( babelConversion.getStatusCode().equals(HttpStatus.MULTIPLE_CHOICES) ) {
+				try {
+					Thread.sleep( 20000 );
```
Comment (tydus101) [inline src/main/java/org/moltimate/moltimatebackend/service/DockingService.java]: Possibly shorten this waiting time? @etiano417

Comment (kdm5498) [reply src/main/java/org/moltimate/moltimatebackend/service/DockingService.java]: In my testing, it normally took at least 30 seconds to convert the file, sometimes up to several minutes, so I think that 20 seconds is a good amount to not spam the openbabel server with requests.

File: src/main/java/org/moltimate/moltimatebackend/dto/request/DockingRequest.java (hunk 1 above)
Comment (kdm5498) [reply src/main/java/org/moltimate/moltimatebackend/dto/request/DockingRequest.java]: Spelling has been corrected.

File: src/main/java/org/moltimate/moltimatebackend/service/DockingService.java (hunk 2 above)
Comment (etiano417) [reply src/main/java/org/moltimate/moltimatebackend/service/DockingService.java]: @tydus101 @kdm5498 the current behavior sounds good to me then. For something like this where its a usability vs performance tradeoff and where we don't have solid numbers on the impact on either in actual use, or what typical user behavior is (say, how may of these can we expect to be converted at a time on average) i'd say whatever your first instinct here is fine. Maybe down the road we can discuss calibrating it, but I wouldn't worry too much about it at this point.

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+autodock.url=http://34.73.73.60:8000/v1/autodock/
+openbabel.url=http://35.245.184.57:8000/v1/obabel/
```
Comment (ada-phillips) [inline src/main/resources/application.properties]: Is there any way we can configure this via maven?

**Pull request merged**